dev-lang/moarvm/moarvm-2022.12.ebuild

```diff
@@ -0,0 +1,58 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+MY_PN="MoarVM"
+if [[ ${PV} == "9999" ]]; then
+	EGIT_REPO_URI="https://github.com/${MY_PN}/${MY_PN}.git"
+	inherit git-r3
+	S="${WORKDIR}/${P}"
+else
+	SRC_URI="http://moarvm.org/releases/${MY_PN}-${PV}.tar.gz"
+	KEYWORDS="~amd64 ~x86"
+	S="${WORKDIR}/${MY_PN}-${PV}"
+fi
+
+DESCRIPTION="A 6model-based VM for NQP and Raku"
+HOMEPAGE="http://moarvm.org"
+LICENSE="Artistic-2"
+SLOT="0"
+
+IUSE="asan clang debug doc +jit optimize static-libs ubsan"
+
+RDEPEND="dev-libs/libatomic_ops
+		>=dev-libs/libuv-1.26:=
+		dev-libs/libffi:="
+DEPEND="${RDEPEND}"
+BDEPEND="clang? ( >=sys-devel/clang-3.1 )
+	dev-lang/perl"
+
+DOCS=( CREDITS README.markdown )
+
+# Tests are conducted via nqp
+RESTRICT="test"
+
+src_configure() {
+	MAKEOPTS+=" NOISY=1"
+	use doc && DOCS+=( docs/* )
+	local prefix="${EPREFIX}/usr"
+	local libdir="${EPREFIX}/usr/$(get_libdir)"
+	einfo "--prefix '${prefix}'"
+	einfo "--libdir '${libdir}'"
+	local myconfigargs=(
+		"--prefix" "${prefix}"
+		"--has-libuv"
+		"--has-libatomic_ops"
+		"--has-libffi"
+		"--libdir" "${libdir}"
+		"--compiler" "$(usex clang clang gcc)"
```

Comment:
Why are you using flags to do that? If you really can't respect CC/CXX, why not use something like `tc-is-clang`? Also, I'm guessing the current logic will fail on clang-only systems w/ `USE=-clang`.

Reply:
I am not using anything. I merely inherited everything from the previous maintainer, Patrick.

There is a saying, if it's not broken, don't fix it.

I will see what I can do about this.

Reply:
I'm trying to address this, but I don't know what exactly you mean by CC, CXX, and `tc-is-clang`.

Do you mean that arguments to --libdir and --prefix should be handled by an existing eclass?

Reply:
Is `tc-is-clang` the right way?

Reply:
There is also a saying "if it's Patrick, assume it's completely broken and wrong".

"The right way" would be to respect CC independently of value. However, I don't know what the option does. Can you tell me that?

Reply:
Are you curious about --compiler option?

Reply:
`perl Configure.pl --help` says
```
    --compiler <compiler>
        Currently supported compilers are "gcc", "clang" and "cl".
```

Reply:
`tc-get-compiler-type` may be able to replace clang USE flag although https://wiki.gentoo.org/wiki/Clang says

> Some packages are aware of the [USE](https://wiki.gentoo.org/wiki/USE_flag)=[clang](https://packages.gentoo.org/useflags/clang) USE flag. This is typically here as specific flags must be used by the package internally for it to compile or work correctly at runtime:

Reply:
I just replaced clang USE flag with `tc-get-compiler-type` in a case statement that sets `compiler` local variable.